pkg/reconciler/v1alpha1/clusteringress/clusteringress.go

```diff
@@ -216,29 +233,47 @@ func (c *Reconciler) reconcile(ctx context.Context, ci *v1alpha1.ClusterIngress)
 	ci.Status.MarkLoadBalancerReady(getLBStatus(gatewayServiceURLFromContext(ctx, ci)))
 	ci.Status.ObservedGeneration = ci.Generation
 
-	// TODO(zhiminx): Istio requires to put certificates under the namespace where Istio ingress
-	// (pods) are deployed so that Istio ingress pods can consume them.
-	// So we need to copy certificates from their origin namespace to the Istio ingress namespace.
-
 	// TODO(zhiminx): currently we turn off Gateway reconciliation as it relies
 	// on Istio 1.1, which is not ready.
 	// We should eventually use a feature flag (in ConfigMap) to turn this on/off.
 
 	if c.enableReconcilingGateway {
+		if !ci.IsPublic() {
+			logger.Infof("ClusterIngress %s is not public. So no need to configure TLS.", ci.Name)
+			return nil
+		}
+
 		// Add the finalizer before adding `Servers` into Gateway so that we can be sure
 		// the `Servers` get cleaned up from Gateway.
 		if err := c.ensureFinalizer(ci); err != nil {
 			return err
 		}
 
-		desiredServers := resources.MakeServers(ci)
-		if err := c.reconcileGateways(ctx, ci, gatewayNames, desiredServers); err != nil {
+		originSecrets, err := resources.GetSecrets(ci, c.secretLister)
+		if err != nil {
+			return err
+		}
+		targetSecrets := resources.MakeSecrets(ctx, originSecrets)
+		if err := c.reconcileCertSecrets(ctx, ci, targetSecrets); err != nil {
 			return err
 		}
+
+		for _, gatewayName := range gatewayNames {
+			ns, err := resources.GatewayServiceNamespace(config.FromContext(ctx).Istio.IngressGateways, gatewayName)
+			if err != nil {
+				return err
+			}
+			desired, err := resources.MakeServers(ci, ns, originSecrets)
```

Comment:
should this origin or target?

Reply:
this is the desired Istio Gateway server, not secret. :)